From ff17fd8ffe1d91bce932af0f8e842e5a643feb66 Mon Sep 17 00:00:00 2001
From: Cupertino Miranda <cmiranda@synopsys.com>
Date: Wed, 5 Apr 2017 14:53:47 +0300
Subject: [PATCH] ldso: Troubleshoot unresolved symbol

New linker is more picky on detection of undefined symbols in dynamic
libraries. The dynamic loader because of some bug in uclibc ends up
having a reference to abort that should not be there.

Signed-off-by: Cupertino Miranda <cmiranda@synopsys.com>
Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
---
 ldso/ldso/ldso.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ldso/ldso/ldso.c b/ldso/ldso/ldso.c
index 4e8a49ef5f91..3e90fc3690f6 100644
--- a/ldso/ldso/ldso.c
+++ b/ldso/ldso/ldso.c
@@ -379,7 +379,8 @@ static struct elf_resolve * add_ldso(struct elf_resolve *tpnt,
 		} else {
 			rpnt = _dl_zalloc(sizeof(struct dyn_elf));
 		}
-		rpnt->dyn = tpnt;
+		if (rpnt)
+			rpnt->dyn = tpnt;
 		tpnt->rtld_flags = RTLD_NOW | RTLD_GLOBAL; /* Must not be LAZY */
 
 	return tpnt;
-- 
2.7.4

